### ISSUE TITLE
VS Code Extension Doesn't Recognize Sapling Repo

### ISSUE THREAD

**Issue by zengk95** [facebook/sapling#282]
When I open up the ISL using VS Code, it gives me on the main screen. 
```
Not a valid repository
c:\Users\kzeng\workspace\crc-worker is not a valid Sapling repository. Clone or init a repository to use ISL.
```
<img width="971" alt="image" src="https://user-images.githubusercontent.com/34172846/204584773-3477db25-5ddc-4605-b7ec-ccab913b4326.png">

I'm using the latest version `Sapling 0.1.20221118-210929-cfbb68aa` on windows using Powershell. I can get sl to work in the web browser and on the console as well. 

Any recommendations?

**Comment by evangrayk** [comment]
Thanks for reporting this! 

If you go to the "output" channel at the bottom, then switch to "Sapling ISL", you'll be able to see the logs from ISL, which should show an error message about why it failed to find a repository.

There's a similar code path here for detecting if the sapling command is not installed correctly. I'm wondering if this is related to "sl" [conflicting with Set-Alias on powershell](https://sapling-scm.com/docs/introduction/installation#windows).

We don't currently have a VS Code config for the extension to set which sl executable name to use, but if this is related to the powershell issue I think that would be a good first step for us to add.

**Comment by zengk95** [comment]
Hmm you're right. 
```
Disposing ISL panel
Populating ISL webview panel
populate isl webview
establish sl client connection for c:\Users\zengk\workspace\test-repo
run command:  sl root
run command:  sl root
run command:  sl config
run command:  sl config
run command:  sl config
Failed to find repository in c:\Users\zengk\workspace\test-repo [Error: Command failed with exit code 1: sl root --noninteractive --config ui.merge=:merge3
'sl' is not recognized as an internal or external command,
operable program or batch file.
	at e.exports (c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:18509)
	at c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:15693
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Fn (c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:85757)
	at async Promise.all (index 0)
	at async kn.getRepoInfo (c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:81984)
	at async c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:87803] {
  shortMessage: 'Command failed with exit code 1: sl root --noninteractive --config ui.merge=:merge3',
  command: 'sl root --noninteractive --config ui.merge=:merge3',
  escapedCommand: 'sl root --noninteractive --config "ui.merge=:merge3"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "'sl' is not recognized as an internal or external command,\r\n" +
    'operable program or batch file.',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
Failed to find repository dotdir in c:\Users\zengk\workspace\test-repo [Error: Command failed with exit code 1: sl root --dotdir --noninteractive --config ui.merge=:merge3
'sl' is not recognized as an internal or external command,
operable program or batch file.
	at e.exports (c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:18509)
	at c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:15693
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Vn (c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:85979)
	at async Promise.all (index 1)
	at async kn.getRepoInfo (c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:81984)
	at async c:\Users\zengk\.vscode\extensions\meta.sapling-scm-0.1.3\dist\extension.js:1:87803] {
  shortMessage: 'Command failed with exit code 1: sl root --dotdir --noninteractive --config ui.merge=:merge3',
  command: 'sl root --dotdir --noninteractive --config ui.merge=:merge3',
  escapedCommand: 'sl root --dotdir --noninteractive --config "ui.merge=:merge3"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "'sl' is not recognized as an internal or external command,\r\n" +
    'operable program or batch file.',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```

**Comment by evangrayk** [comment]
dce1ec6f1d37 updates our logic to try `sl.exe` on windows, and also adds a vscode setting so you could customize this to a specific installation location, hopefully that should help. I'll do a new vscode extension release with this change.

**Comment by evangrayk** [comment]
I'll also add a change so that we detect this as sl not being found on the PATH rather than as an invalid repo. Looks like our ENOENT check doesn't work on windows.

**Comment by evangrayk** [comment]
A new version (0.1.5) of the vscode extension has been published with these fixes. Let us know if this helps, or if you still see problems with this!

**Comment by bolinfest** [comment]
> The only issue was that the open link for the installation docs didn't work in the error screen (maybe a VS Code issue)

which error screen?

> and I had to add the sapling directory to my environment variables in windows

You mean you had to add the folder with `sl.exe` to your `%PATH%`?

**Comment by zengk95** [comment]
This error screen 
![image](https://user-images.githubusercontent.com/34172846/205695315-8e3f6927-f65b-49e3-9886-31eb264a88d8.png)

The button didn't work for me.

And yea had to add the folder with sl.exe into the environment variables through the GUI in windows.

**Comment by bolinfest** [comment]
Ah, so I believe the logic for the button is here:

https://github.com/facebook/sapling/blob/598529957ba6e9f4c559c4b7e50a242b38c00899/addons/isl/src/App.tsx#L129-L137

The client portion of `openExternalLink()` is here:

https://github.com/facebook/sapling/blob/598529957ba6e9f4c559c4b7e50a242b38c00899/addons/vscode/webview/vscodeWebviewPlatform.ts#L28-L30

which should be handled here:

https://github.com/facebook/sapling/blob/598529957ba6e9f4c559c4b7e50a242b38c00899/addons/vscode/extension/vscodePlatform.ts#L37-L40

so not sure where things are going wrong.

**Comment by bolinfest** [comment]
I just published v0.1.6 of the VS Code extension, which includes the fix for the "See installation docs" button not working: https://github.com/facebook/sapling/commit/08df5afedd5cc6d9e55071070f381766238e7bd6. That turned out to be a subtle issue, so thanks for reporting, @zengk95!
